gdal-sys/src/ogr.rs

```diff
@@ -42,10 +49,17 @@ extern {
     pub fn OGR_G_DestroyGeometry(hGeom: *mut c_void);
     pub fn OGR_Fld_GetNameRef(hDefn: *const c_void) -> *const c_char;
     pub fn OGR_Fld_GetType(hDefn: *const c_void) -> c_int;
+    pub fn OGR_Fld_Create(pszName: *const c_char, eType: c_int) -> *const c_void;
+    pub fn OGR_Fld_GetWidth(hDefn: *const c_void) -> c_int;
+    pub fn OGR_Fld_GetPrecision(hDefn: *const c_void) -> c_int;
+    pub fn OGR_Fld_SetWidth(hDefn: *const c_void, nNewWidth: c_int) -> c_void;
+    pub fn OGR_Fld_SetPrecision(hDefn: *const c_void, nNewPrecision: c_int) -> c_void;
+    pub fn OGR_Fld_Destroy(hDefn: *mut c_void) -> c_void;
     pub fn OGRFree(ptr: *mut c_void);
     pub fn VSIFree(ptr: *mut c_void);
 }
 
+pub const OFT_INTEGER:            c_int = 0;
```

Comment:
I think it would be nice to change the OFT types into an enum and move it into ogr_enums.rs. This way all the FFI methods could take OgrFieldType instead of c_int as parameter. ([link to the OGR enum](http://www.gdal.org/ogr__core_8h.html#a787194bea637faf12d61643124a7c9fc))